Revise example links and integration test references (#10621)
ROCm/rocm-libraries: projects/hipdnn/docs/user-guides/how-to/develop-plugins.rst

Reviewer: ```suggestion
Integration tests validate end-to-end functionality of your plugin. There are currently two categories of integration tests, internal and external.

Internal integration tests are run as part of the plugin's own test suite:

Reviewer: ```suggestion
  - Tests are divided into two categories designated by the prefix argument passed to ``INSTANTIATE_TEST_SUITE_P``.
   
The internal integration tests are typically simple tests to ensure that the plugin is able to properly load and run kernels on GPU hardware. Integrations tests for numerical accuracy are better handled using the external integration tests (below).
```

- Validating and documenting supported operations, hardware requirements, and limitations.
- Including unit tests and integration tests.

Example engine plugin implementation
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> A fully functional example of a hipDNN engine plugin is available `here <https://github.com/ROCm/rocm-libraries/blob/develop/projects/hipdnn/samples/example_engine_plugin/README.md>`_

Author: ```suggestion
A fully functional example of a hipDNN engine plugin is available `here <https://github.com/ROCm/rocm-libraries/blob/develop/projects/hipdnn/samples/example_engine_plugin/README.md>`_.
```

    - **Smoke**: These tests are designed to test features using the smallest possible shape and run quickly (the combined smoke test run time must be under 5 mins).
    - **Full**: These tests can contain regression shapes, large shapes, or slow shapes.

> For a comprehensive example of an integration test, see `Integration Tests <https://github.com/ROCm/rocm-libraries/blob/develop/dnn-providers/integration-tests/README.md>`_ in the hipDNN repo.

Reviewer: ```suggestion
External integrations tests use an external integration test executable written to load plugins and perform end-to-end verification of graph operations using the plugin. For details on how to use the external integration test harness see `Integration Tests <https://github.com/ROCm/rocm-libraries/blob/develop/dnn-providers/integration-tests/README.md>`_ in the hipDNN repo.
```
